src/Microsoft.ML.Transforms/Text/LdaTransform.cs

```diff
@@ -930,46 +657,475 @@ public void Dispose()
             }
         }
 
-        private ColumnType[] InitColumnTypes(int numTopics)
+        private sealed class Mapper : MapperBase
         {
-            Host.Assert(Utils.Size(Infos) > 0);
-            var types = new ColumnType[Infos.Length];
-            for (int c = 0; c < Infos.Length; c++)
-                types[c] = new VectorType(NumberType.Float, numTopics);
-            return types;
+            private readonly LdaTransformer _parent;
+            private readonly ColumnType[] _srcTypes;
+            private readonly int[] _srcCols;
+
+            public Mapper(LdaTransformer parent, Schema inputSchema)
+                : base(parent.Host.Register(nameof(Mapper)), parent, inputSchema)
+            {
+                _parent = parent;
+                _srcTypes = new ColumnType[_parent.ColumnPairs.Length];
+                _srcCols = new int[_parent.ColumnPairs.Length];
+
+                for (int i = 0; i < _parent.ColumnPairs.Length; i++)
+                {
+                    if (!inputSchema.TryGetColumnIndex(_parent.ColumnPairs[i].input, out _srcCols[i]))
+                        throw Host.ExceptSchemaMismatch(nameof(inputSchema), "input", _parent.ColumnPairs[i].input);
+
+                    var srcCol = inputSchema[_srcCols[i]];
+                    if (!srcCol.Type.IsKnownSizeVector || !(srcCol.Type.ItemType is NumberType))
+                        throw Host.ExceptSchemaMismatch(nameof(inputSchema), "input", _parent.ColumnPairs[i].input);
```

Comment:
>ExceptSchemaMismatch [](start = 35, length = 20)

wrong error message. It will say that the column is missing, but it should say that the type is not an expected type